code/_core/obj/structure/totem.dm

```diff
@@ -0,0 +1,243 @@
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.health.stamina_current >= L.health.stamina_max)
+			continue
+		if(L.health.get_stamina_loss())
+			L.stamina_regen_buffer += (3 + (3 * leveled_effect))
+		var/obj/effect/temp/healing/H = new(L.loc,10,COLOR_GREEN)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/stamina_deal
+	name = "totem of stamina degeneration"
+	desc_extended = "A totem that will damage the caster's enemies' stamina."
+	icon_state = "staminaloss"
+
+/obj/structure/totem/stamina_deal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag == affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		L.stamina_regen_buffer -= (3 + (3 * leveled_effect))
+		var/obj/effect/temp/electricity/H = new(L.loc,10,COLOR_GREEN)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/mana_heal
+	name = "totem of mana regeneration"
+	desc_extended = "A totem that will restore the caster's and their allies' mana."
+	icon_state = "mana"
+
+/obj/structure/totem/mana_heal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.health.mana_current >= L.health.mana_max)
+			continue
+		if(L.health.get_mana_loss())
+			L.mana_regen_buffer += (3 + (3 * leveled_effect))
+		var/obj/effect/temp/healing/H = new(L.loc,10,COLOR_BLUE)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/mana_deal
+	name = "totem of mana regeneration"
+	desc_extended = "A totem that will damage the caster's enemies' mana."
+	icon_state = "manaloss"
+
+/obj/structure/totem/mana_deal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag == affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		L.mana_regen_buffer -= (3 + (3 * leveled_effect))
+		var/obj/effect/temp/electricity/H = new(L.loc,10,COLOR_BLUE)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/sacred_flame
+	name = "totem of sacred flame"
+	desc = "Someone thought to take a book and stick it inside the totem."
+	desc_extended = "A totem that will set ablaze the caster's enemies."
+	icon_state = "flame"
+
+/obj/structure/totem/sacred_flame/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag == affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		L.ignite(SECONDS_TO_DECISECONDS(1.5))
+		var/obj/effect/temp/electricity/H = new(L.loc,10,COLOR_RED)
```

Comment:
You should use CREATE(/obj/effect/temp/healing/,get_turf(L)) instead and remove the INITIALIZE(H) and friends.